compaction/compaction_manager.cc

```diff
@@ -1133,17 +1148,20 @@ class compaction_manager::rewrite_sstables_compaction_task : public compaction_m
 
 protected:
     virtual future<> do_run() override {
+        compaction_task_stats cts{};
         switch_state(state::pending);
         auto maintenance_permit = co_await acquire_semaphore(_cm._maintenance_ops_sem);
 
         while (!_sstables.empty() && can_proceed()) {
             auto sst = consume_sstable();
-            co_await rewrite_sstable(std::move(sst));
+            co_await rewrite_sstable(std::move(sst), cts);
```

Comment:
it'd be cleaner if rewrite_sstable will just return the `compaction_result`
and the accumulation into `compaction_task_stats` will be done here, by the caller.